central/complianceoperator/v2/profiles/service/service_impl.go

```diff
@@ -79,26 +75,37 @@ func (s *serviceImpl) GetComplianceProfile(ctx context.Context, req *v2.Resource
 }
 
 // ListComplianceProfiles returns profiles matching given query
-func (s *serviceImpl) ListComplianceProfiles(ctx context.Context, query *v2.RawQuery) (*v2.ListComplianceProfilesResponse, error) {
-	// Fill in Query.
-	parsedQuery, err := search.ParseQuery(query.GetQuery(), search.MatchAllIfEmpty())
-	if err != nil {
-		return nil, errors.Wrapf(errox.InvalidArgs, "Unable to parse query %v", err)
+func (s *serviceImpl) ListComplianceProfiles(ctx context.Context, request *v2.ProfilesForClusterRequest) (*v2.ListComplianceProfilesResponse, error) {
+	if request.GetClusterId() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "cluster is required")
 	}
 
-	// Fill in pagination.
-	paginated.FillPaginationV2(parsedQuery, query.GetPagination(), maxPaginationLimit)
```

Comment:
Do we not need to do this anymore?

Reply:
Probably but I'm tired of this PR being open.  So https://issues.redhat.com/browse/ROX-21977